src/main/org/deidentifier/arx/ARXConfiguration.java

```diff
@@ -707,7 +707,9 @@ public Monotonicity getMonotonicityOfPrivacy() {
      * @return
      */
     public Monotonicity getMonotonicityOfUtility() {
-        if (metric.isMonotonic() || (this.getMaxOutliers() == 0d) || this.isPracticalMonotonicity()) {
+        if ((metric.isMonotonicWithGeneralization() && this.getMaxOutliers() == 0) || 
```

Comment:
It would probably be a good idea to take a look at this in more detail. How did the "bungled logic" look like? Which comparison failed but was not reflected by the unit tests? Can you provide me with code?